### PR TITLE
LPS-177017 Update event input component

### PR DESCRIPTION
**Pull request by p2kmgcl**
Motivation
=======
Update EventInput component to use DateInput, which enables setting a Date/Date Range in segment Events. https://issues.liferay.com/browse/LPS-177017

Proposed Solution
========
We have added two new inputs to the Event node. The first one allows selecting a date `operatorName`, being one of `SUPPORTED_EVENT_DATE_OPERATORS`. If none of them is selected, we use `ever` as default value, as it means that this operator will not be taken into account.

If the user selects some other operator, then we show a `DateInput` component next to it, so they can select a Date Range (for `between` operator) or a single Date.

Steps to Verify:
----------------
1. Create some segment with an Event node in AC
2. Connect portal to AC
3. Sync AC segments with portal
4. In portal, edit the created segments
5. Check that the two new inputs appear

Screenshots (optional):
-----------------------

image
image
image

### CODE REVIEW

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js [hunk 1]
```diff
@@ -57,6 +62,13 @@ function EventInput({
 		(operator) => operator.name === criterion.operatorName
 	)?.name;
 
+	const showDateSelector = [
+		DATE_OPERATORS.BETWEEN,
+		RELATIONAL_OPERATORS.EQ,
+		RELATIONAL_OPERATORS.GT,
+		RELATIONAL_OPERATORS.LT,
+	].includes(criterion.day.operatorName);
```
Comment (claraizquierdo) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js:70]: Hello @p2kmgcl! The day field will be undefined in the 'ever' date modifier so, accessing day.x might fail.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js [hunk 2]
```diff
@@ -116,6 +128,43 @@ function EventInput({
...
+							day: {
+								...criterion.day,
+								operatorName: event.target.value,
+							},
+						})
+					}
+					options={SUPPORTED_EVENT_DATE_OPERATORS.map(
+						({label, name: value}) => ({label, value})
+					)}
+					value={criterion.day.operatorName}
```
Comment (claraizquierdo) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js:149]: I think that this value will not work with the ever modifier 🤔

**Review by claraizquierdo — CHANGES_REQUESTED** [review]
Hi @p2kmgcl ,
I left two comments. I did not  have time to see it working but having a look at the code I think that we could have problems with the ever date modifier 🤔  What do you think?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js (hunk 2 above)
Comment (p2kmgcl) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js:149]: :thinking: I need to have a look. I think I saw it working but I am probably wrong. I'll check that :smile:

**Comment by p2kmgcl** [comment]
Hey @claraizquierdo, I have updated the component checking that `criterion.day` exists everywhere. I have also removed the `defaultValue` property and replaced it with `value={criterion.day?.operatorName || DATE_OPERATORS.EVER` because React thought I was mixing controlled/uncontrolled components.

Now it should work perfectly :smile:

**Comment by p2kmgcl** [comment]
ci:test:sf

**Comment by p2kmgcl** [comment]
ci:test:tango

**Comment by p2kmgcl** [comment]
ci:test:relevant

**Comment by p2kmgcl** [comment]
ci:test:tango

**Comment by p2kmgcl** [comment]
ci:test:tango

**Comment by Stephy6** [comment]
Hi @p2kmgcl ! it works perfectly but I found a strange behaviour: when I select the between operator, select a range and then I change the time operator to since(for example) the range value is kept in the input:

Screenshot 2023-03-07 at 15 22 41

I'm not sure if that should behave that way or not.

**Comment by p2kmgcl** [comment]
> I'm not sure if that should behave that way or not.

You're right, the behavior when changing from/to `between` is a bit weird. What do you think about emptying the field when this happens? I can just set the value to empty string when this changes, or maybe change it to today's date.

**Comment by Stephy6** [comment]
> > I'm not sure if that should behave that way or not.
> 
> You're right, the behavior when changing from/to `between` is a bit weird. What do you think about emptying the field when this happens? I can just set the value to empty string when this changes, or maybe change it to today's date.

In my opinion it should be a value inside like today's date but I'm not sure.

**Comment by p2kmgcl** [comment]
I think we might keep the start value, this is a bit arbitrary but it is start or end :sweat_smile:. I am going to do the test and check how it work.

**Review by claraizquierdo — CHANGES_REQUESTED** [review]
Hi @p2kmgcl ,

I left a few comments, let me know if you have any doubt or if I can help with something. Also, if you could set the widths of the inputs to be a little bit more similar to de designs could be awesome :)  https://www.figma.com/file/4pdAjSeb4L89hptsbvxzNG/LPS-172194---Edit-AC-Event-Segments-in-DXP?node-id=3-4&t=YvA2y0o9C1GCO0om-0

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js [hunk 3]
```diff
@@ -118,6 +122,42 @@ function EventInput({
...
+							},
+						})
+					}
+					options={SUPPORTED_EVENT_DATE_OPERATORS.map(
+						({label, name: value}) => ({label, value})
+					)}
+					value={criterion.day?.operatorName || DATE_OPERATORS.EVER}
+				/>
+
+				{criterion.day?.operatorName !== DATE_OPERATORS.EVER ? (
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js:145]: I think we need to check both things: it is not undefined and also it is not the 'ever' string... or do we completely remove the `criterion.day` when 'ever' is selected?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js [hunk 4]
```diff
@@ -118,6 +122,42 @@ function EventInput({
...
+							day: {
+								...criterion.day,
+								operatorName: event.target.value,
+							},
+						})
+					}
+					options={SUPPORTED_EVENT_DATE_OPERATORS.map(
+						({label, name: value}) => ({label, value})
+					)}
+					value={criterion.day?.operatorName || DATE_OPERATORS.EVER}
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js:142]: aaaaamiga, thanks so much for that, I'll fix it

**Comment by p2kmgcl** [comment]
I've just included some changes, but will continue with it tomorrow morning :smile:

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js:142]: No problem, sorry if we didn't provide good requirements 😞

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js]: The `criterion.day` will not exists for the 'ever' operator.

**Comment by p2kmgcl** [comment]
@claraizquierdo I just applied all the changes :smile:. finally I managed the "since" operator behavior inside the component itself. I tried to move it outside of the component, but it was a bit of a mess, and I needed to refactor a lot of code :sweat: 

Anyway, now it works :partying_face: 

Speaking about the DateInput not being updated (what @Stephy6 mentioned), I need to update it to update the displayed date when the provided value changes. I am going to add a commit :smile:

**Comment by p2kmgcl** [comment]
done!

**Review by claraizquierdo — APPROVED** [review]
LGTM! 🎉

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Review by claraizquierdo — APPROVED** [review]
Everything is still working 🎉

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:forward

**Pull request closed without merging**